pkgs/development/python-modules/python-imports-check-libstdcxx/test_forward_compatibility_fails.py

```diff
@@ -0,0 +1,39 @@
+import os
+import sys
+
+assert (
+    "EXPECTED_LIBSTDCXX" in os.environ
+), "The derivation failed to export EXPECTED_LIBSTDCXX"
+
+expected_libstdcxx_root = os.environ["EXPECTED_LIBSTDCXX"]
+
+import load_libstdcxx_old
+import psutil
+
+try:
+    import load_libstdcxx_new
+except ImportError:
+    print(f"python-imports-check-load-libstdcxx successfuly fails", file=sys.stderr)
+    sys.exit(0)
```

Comment:
perhaps i'm missing something, but what is the value in checking if an import fails? don't we want to be checking that imports like [this one](https://github.com/NixOS/nixpkgs/issues/221739#issuecomment-1474754347) succeed?

Reply:
> checking that imports like this one succeed?

What we want to check is that when we do such an import we'll have the new libstdc++ loaded, and not one from gcc11.

Thus the idea with appending this module to the `pythonCheckImports` is that if we have accidentally linked something against the old libstdc++, we'll fail right away and not even try reverse dependencies

For example, right now we can build jaxlib just fine, but at runtime we find it doesn't work with jax. With this module in imports check, jaxlib will fail at build time